modern/src/settings/UserPage.jsx

```diff
@@ -267,10 +267,11 @@ const UserPage = () => {
                 </Select>
               </FormControl>
               <SelectField
-                value={(item.attributes && item.attributes.timezone) || ''}
-                emptyValue=""
+                value={(item.attributes && item.attributes.timezone) || null}
+                emptyValue={null}
                 onChange={(e) => setItem({ ...item, attributes: { ...item.attributes, timezone: e.target.value } })}
                 endpoint="/api/server/timezones"
+                keyField={false}
```

Comment:
Why do we need `keyField` in addition to `keyGetter`?

Reply:
Because sometimes we have to find the default/selected option even if only the key value is passed. For example:   https://github.com/traccar/traccar-web/blob/2bd044ecf9fa456e168057acc513bd3b4582d0c6/modern/src/settings/PreferencesPage.jsx#L268

Reply:
Unfortunately it does not answer my question.

Reply:
Ok, let me try to explain it in a different way `value` in the above linked case is either a full Object, from which we can get the label via `titleGetter` or a string `'name'`. We need to have a way to get a label from the value `'name'` and we cannot pass a string to `titleGetter`. The current solution searches all the possible options (so an array of Objects) for the value `'name'` and it needs to know in which property of the Object to look for. `keyGetter` gets the key value from the Object and cannot help us to search for the appropriate Object in the array of Objects.

The second option would be to change the code so that the value passed is always a full object.

Reply:
Oh, and the original example you commented it flips it around, as when we add the above logic, we need a way to handle non Object values (so just an array of strings). So indeed, this can all be simplified if we change the few instances where the default value passed is of a different type (so a string instead of Object) as all the options.

Reply:
I all for simplification if there's an opportunity for it.

Reply:
So apparently, after reading the above mentioned issue and this one (https://github.com/mui/material-ui/issues/23708)  I found out that my initial explanation was a bit to narrow. Because we are always working with **keys/IDs** (eg: supplied `value` is always a key/id, no matter if it's a default value or one from the backend, and we are always returning key/ids in onChange) we need to convert that to full Object(s) if we wish to use `titleGetter` - contrary to what I wrote before, that this is not only needed for default values but we need it for stored/selected values too.
Because the flow (for a selected or default) goes like that:
`value` -> `getOptionLabel` -> `titleGetter` for which we need the Object. So there are basically 2 options:
1. We convert the `value` key/id to an Object and feed that to `titleGetter` - like we do now, and where we need the `keyField` to search the array for the Object
2. We always work with Objects instead of keys/ids -> so `value` would contain Object(s) and `onChange` would return Objects and the Redux would keep objects instead of keys/ids - **I think this would be way too much work**.

Unfortunately this works differently than `Select` and causes us problems. Also doesen't seem that the MUI developers will address this, as people have been complaining about this for quite some time with no proper solution in sight. We also can't easly rework this to use something akin like `objectGetter` because it won't work for `endpoint` enabled `SelectField`. 
TL;DR: I have not found any other proper solutions for this and this implementation - even tho it's not so nice, is the best one we can do without major refactoring of either the frontend or the backend.

Reply:
There must be a way to wrap `Autocomplete` with something to make it work similar to normal select field.

Reply:
I mean, that's exactly what `SelectField` now does, but if we don't have a uniform property in all of the Objects for where the `value` is (eg. `id`) we need to specify where our solution/wrapper searches for Object(s). Similarly why we need `keyGetter`, `titleGetter` because Objects do not have uniform properties (eg. value in `id`, title/label in `name`) we need a way to specify this. 
On the other hand, I did just figure out I am a little bit of an idiot and I can probably reuse `keyGetter` now with all the info I have and the refactoring.

Reply:
That was my original point. Having key and title getters should be enough.

Reply:
Yeah I over complicated and did not re-think completely. This latest commit is a lot simpler and works good.